### PR TITLE
Add group trigger update functionality to the module

### PR DESCRIPTION
**Pull request by dkorn**

### CODE REVIEW

File: library/hawkular_alerts_group.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 DOCUMENTATION = '''
 ---
 module: hawkular_alerts_group
-short_description: Creating Group Triggers in Hawkular Alerting
+short_description: Creating and updatingGroup Triggers in Hawkular Alerting
```
Comment (cben) [inline library/hawkular_alerts_group.py]: s/updatingGroup/updating Group/

File: library/hawkular_alerts_group.py [hunk 2]
```diff
@@ -89,42 +89,90 @@
...
+
+    def create_or_update_group_trigger(self, name, group_id, severity, enabled):
+        """ Creates or updates a group trigger in Hawkular Alerts
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        try:
+            gt = self.client.get_trigger(group_id)
```
Comment (cben) [inline library/hawkular_alerts_group.py]: I think it'd be more precise to have only this line inside the `try..except`, moving `required_updates()` and  `update_group_trigger()` code below.

File: library/hawkular_alerts_group.py [hunk 3]
```diff
@@ -89,42 +89,90 @@
...
+        """ Checks whether an update is required for the group trigger
+
+            Returns:
+                True if the passed trigger name, severity or enabled status
+                are different from the ones set in the trigger, False otherwise
+        """
+        updates = {}
+        if (name and trigger.name != name):
+            updates["name"] = name
+        if (severity and trigger.severity != severity):
```
Comment (cben) [inline library/hawkular_alerts_group.py]: Does this work?
I suspect `trigger.severify` is already the client's representation eg. `Severity.LOW` (which happens to equal "LOW" but we shouldn't assume that), while `severity` on right side is whatever the module param was e.g. "low"?

File: library/hawkular_alerts_group.py [hunk 4]
```diff
@@ -89,42 +89,90 @@
...
+            updates["severity"] = severity.upper()
```
Comment (cben) [inline library/hawkular_alerts_group.py]: I think this should use `getattr(hawkular.alerts.Severity, severity.upper())`.

Or better, do that translation at higher level — in `create_or_update_group_trigger()` or even `main()` and then create/updating methods could just assign/compare severities as-is.

**Review by cben — COMMENTED** [review]
Looks good overall.

Can you declare `choices` on severity module param?

File: library/hawkular_alerts_group.py (hunk 1 above)
Comment (dkorn) [reply library/hawkular_alerts_group.py]: Done, thanks

File: library/hawkular_alerts_group.py (hunk 2 above)
Comment (dkorn) [reply library/hawkular_alerts_group.py]: Done, thanks

File: library/hawkular_alerts_group.py (hunk 4 above)
Comment (dkorn) [reply library/hawkular_alerts_group.py]: Done, thanks

File: library/hawkular_alerts_group.py (hunk 3 above)
Comment (dkorn) [reply library/hawkular_alerts_group.py]: embraced you suggestion below, thanks

**Comment by dkorn** [comment]
@cben addressed all comments, any others?
@enoodle PTAL

File: library/hawkular_alerts_group_trigger.py [hunk 5]
```diff
@@ -122,37 +122,85 @@ def delete_group_trigger(self, group_id):
         except Exception as e:
             self.module.fail_json(msg="Failed to delete group trigger. Error: {error}".format(error=e))
 
-    def create_group_trigger(self, name, group_id, severity, enabled):
-        """ Creates a group trigger in Hawkular Alerting component
+    def required_updates(self, trigger, name, severity, enabled):
+        """ Checks whether an update is required for the group trigger
+
+            Returns:
+                True if the passed trigger name, severity or enabled status
```
Comment (cben) [inline library/hawkular_alerts_group_trigger.py]: wrong, returns a dict

File: library/hawkular_alerts_group_trigger.py [hunk 6]
```diff
@@ -122,37 +122,85 @@ def delete_group_trigger(self, group_id):
...
+                are different from the ones set in the trigger, False otherwise
+        """
+        updates = {}
+        if (name and trigger.name != name):
```
Comment (cben) [inline library/hawkular_alerts_group_trigger.py]: the left half is to allow omitted module params to not modify the trigger, right?

can any of those be 0, "", or False? (especially `enabled: false` in yaml?)
I think in that case this would erronously decide they don't need updating.

File: library/hawkular_alerts_group_trigger.py [hunk 7]
```diff
@@ -122,37 +122,85 @@ def delete_group_trigger(self, group_id):
...
+            updates["name"] = name
+        if (severity and trigger.severity != severity):
+            updates["severity"] = severity
+        if (enabled and trigger.enabled != enabled):
+            updates["enabled"] = enabled
+        return updates
+
+    def update_group_trigger(self, trigger, group_id, updates):
```
Comment (cben) [inline library/hawkular_alerts_group_trigger.py]: I believe you could drop `group_id` param, fetch `trigger.id` where needed.
Not sure why it sounds better to me, your call.

File: library/hawkular_alerts_group_trigger.py (hunk 5 above)
Comment (dkorn) [reply library/hawkular_alerts_group_trigger.py]: Done, thanks

File: library/hawkular_alerts_group_trigger.py (hunk 6 above)
Comment (dkorn) [reply library/hawkular_alerts_group_trigger.py]: right.
That's correct, what about the following check?
```python 
name is not None
```
@cben

**Comment by cben** [comment]
>
> name is not None
>
>
That should work.  I'd add a comment explaining why we're testing None.
Do you have any tests for update with omitted params?  Oh, I see, we don't
have any tests, period :-)

File: library/hawkular_alerts_group_trigger.py (hunk 7 above)
Comment (dkorn) [reply library/hawkular_alerts_group_trigger.py]: Done, thanks

**Comment by dkorn** [comment]
> That should work.  I'd add a comment explaining why we're testing None.

Added

> Do you have any tests for update with omitted params?  Oh, I see, we don't
> have any tests, period :-)

WIP :)

**Comment by cben** [comment]
LGTM
